server/app/assets/javascripts/file_upload.ts

```diff
@@ -53,6 +56,62 @@ export function init() {
       blockForm.submit()
     }
   })
+
+  const uploadedFilesAttribute = blockForm
+    .querySelector(`[${UPLOADED_FILE_ATTR}]`)
+    ?.getAttribute(UPLOADED_FILE_ATTR)
+
+  if (uploadedFilesAttribute) {
+    const uploadedFilesArray = JSON.parse(uploadedFilesAttribute) as string[]
+    blockForm.addEventListener('formdata', (event) => {
+      const formData = event.formData
+      const file = formData.get('file') as File
+
+      const newName = getUniqueName(file.name, uploadedFilesArray)
+      if (file.name != newName) {
+        // Rename uploaded file, if a file with the same name has already been uploaded.
+        formData.delete('file')
+        formData.append('file', file, newName)
+      }
+    })
+  }
+}
+
+// Exporting so we can test this.
```

Comment:
**suggestion:** Could we add a TSDoc with a summary of what this function does, since its a bit complicated. And / or feel free to add some more comments in the function itself.

Reply:
done.